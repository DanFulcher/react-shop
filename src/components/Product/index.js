import React from 'react';
import PropTypes from 'prop-types';
import { Col } from 'react-bootstrap';

import './index.scss';

const Product = ({
  id, img, title, desc, price, addToCart,
}) => {
  const handleClick = () => {
    addToCart(id);
  };
  return (
    <Col className="products__product" md={4}>
      <img src={img} alt={title} />
      <h3>{title}</h3>
      <div className="buttonHolder">
        <button type="button" className="button addToCart" onClick={handleClick} />
      </div>
      <p className="desc">{desc}</p>
      <p className="price">
        <b>
          {`Â£${price}`}
        </b>
      </p>
    </Col>
  );
};


Product.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  addToCart: PropTypes.func.isRequired,
  img: PropTypes.string,
  desc: PropTypes.string,
};

Product.defaultProps = {
  img: undefined,
  desc: undefined,
};

export default Product;
