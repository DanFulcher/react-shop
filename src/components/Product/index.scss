.products {
    &__product {
        
        margin-bottom: 40px;
        img {
            width:100%;
            margin-bottom:5px;
        }
        .desc {
            width:90%;
        }
        .price {
            margin-bottom:20px;
        }
        .buttonHolder {
            display: flex;
            justify-content: flex-end;
        }
        .addToCart {
            cursor:pointer;
            background: #ff7979;
            border:1px solid #ff7979;
            padding:10px;
            border-radius:50%;
            width:45px;
            height:45px;
            position: relative;
            &::before, &::after {
                content:'';
                width:40%;
                height:2px;
                background:#fff;
                position: absolute;
                top:50%;
                left:50%;
            }
            &::before {
                transform: translate(-50%, -50%);
            }
            &::after {
                transform: translate(-50%, -50%) rotate(-90deg);
            }
            &:hover {
                background:#fff;
                &::before, &::after {
                    background:#ff7979;
                }
            }
        }
    }
}