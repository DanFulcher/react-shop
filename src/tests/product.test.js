import React from 'react';
import Product from '../components/Product';
import renderer from 'react-test-renderer';

import { dispatch } from 'react-redux';


const item = {id:1,title:'Winter body', desc: "Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima, ex.", price:110,img:'../images/item1.jpg'};

describe('<Product /> functions', () => {
    it('tests the Add To Cart function', () => {
        const wrapper = renderer.create(<Product item={item} addToCart={dispatch(addToCart(id))}  />);
        const inst = wrapper.getInstance();
        expect(inst.handleClick('1')).toMatchSnapshot();
    });
})