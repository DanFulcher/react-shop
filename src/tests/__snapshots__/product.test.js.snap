// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product /> functions tests the Add To Cart function 1`] = `
<div
  className="products__product col-md-4"
>
  <img
    alt="Winter body"
    src="../images/item1.jpg"
  />
  <h3>
    Winter body
  </h3>
  <div
    className="buttonHolder"
  >
    <button
      className="button addToCart"
      onClick={[Function]}
      type="button"
    />
  </div>
  <p
    className="desc"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima, ex.
  </p>
  <p
    className="price"
  >
    <b>
      Â£110
    </b>
  </p>
</div>
`;
